genet/core.py

```diff
@@ -2263,3 +2264,48 @@ def get_link_slope_dictionary(self, elevation_dict):
             slope_dict[link_id] = {'slope': link_slope}
 
         return slope_dict
+
+    def split_link_at_point(self, link_id, point_lat, point_lon, new_node_id):
```

Comment:
probably safer to remove the user defined node and use https://github.com/arup-group/genet/blob/4fcc3d36a49f8b042e7987082802ee61f3c2fc34/genet/core.py#L1919 to find a unique node ID that doesn't already exist in the network. Otherwise you'll have to warn the user of changing node IDs